alphacsc/_encoder.py

```diff
@@ -0,0 +1,336 @@
+import numpy as np
+from .loss_and_gradient import compute_X_and_objective_multi
+from .update_z_multi import update_z_multi
+from .utils import check_dimension, lil
+
+
+# XXX check consistency / proper use!
+def get_z_encoder_for(
+        solver,
+        z_kwargs,
+        X,
+        D_hat,
+        n_atoms,
+        atom_support,
+        algorithm,
+        reg,
+        loss,
+        loss_params,
+        uv_constraint,
+        feasible_evaluation,
+        n_jobs,
+        use_sparse_z):
+    """
+    Returns a z encoder for the required solver.
+
+    Parameters
+    ----------
+    solver : str
+        The solver to use for the z update. Options are
+        {{'l_bfgs' | 'lgcd'}}.
+    z_kwargs : dict
+        Additional keyword arguments to pass to update_z_multi.
+    X : array, shape (n_trials, n_channels, n_times)
+        The data on which to perform CSC.
+    D_hat : array, shape e(n_trials, n_channels, n_times) or
```

Comment:
```suggestion
    D_hat : array, shape (n_trials, n_channels, n_times) or
```
also I doubt the line below renders fine with numpydoc / sphinx